### PR TITLE
caddy1: remove

### PR DESCRIPTION
**Pull request by zowoq**
###### Motivation for this change

https://github.com/caddyserver/caddy/blob/master/.github/SECURITY.md#supported-versions

Caddy 1.x is unsupported.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zowoq** [comment]
@ofborg test caddy

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -67,6 +67,7 @@ mapAliases ({
   buildGo112Package = throw "buildGo112Package has been removed"; # added 2020-04-26
   buildGo112Module = throw "buildGo112Module has been removed"; # added 2020-04-26
   bundler_HEAD = bundler; # added 2015-11-15
+  caddy1 = throw "caddy 1.x has been removed from nixpkgs, as it was unmaintained: https://github.com/caddyserver/caddy/blob/master/.github/SECURITY.md#supported-versions"; # added 2020-10-02
```
Comment (doronbehar) [inline pkgs/top-level/aliases.nix:70]:
```suggestion
  caddy1 = throw "caddy 1.x has been removed from nixpkgs, as it is unmaintained: https://github.com/caddyserver/caddy/blob/master/.github/SECURITY.md#supported-versions"; # added 2020-10-02
```

**Review by doronbehar — COMMENTED** [review]
Diff LGTM.

**Review by doronbehar — APPROVED** [review]

File: pkgs/top-level/aliases.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/top-level/aliases.nix]: Auto line break?

**Pull request merged**